### ISSUE TITLE
Update some versions and get rid of warnings

### ISSUE THREAD

**Issue by maf-soft** [mariomeyrelles/bicep-functions-cosmos-keyvault#1]
Hi, thanks a lot for your great work. I had to change a few things to get it to work and I also changed some things to get rid of all warnings. Should I contribute my changes?

**Comment by mariomeyrelles** [comment]
Hello @maf-soft, I would certainly be happy with your changes. Probably, some changes have happened since I wrote this article, and of course, there can always be good suggestions from the readers :)

**Comment by maf-soft** [comment]
Wow, that was fast :) unfortunately, I don't really know what's the best way.

**Comment by mariomeyrelles** [comment]
Do you think you can create a Pull Request?

**Comment by maf-soft** [comment]
Normally I know how, but I couldn't imagine that it's possible without asking you first :)

**Comment by maf-soft** [comment]
We should really delete this ticket later :-D - I didn't know how to contact you otherwise.

**Comment by mariomeyrelles** [comment]
I liked the ticket approach and I believe we can leave as it its. I will close when your code arrives.

**Comment by maf-soft** [comment]
Hmm, so everyone can read my silly questions. Will create a fork now, or is there an easier way?

**Comment by maf-soft** [comment]
Maybe now is an opportunity for you to go back over your code and also find things you would do differently these days, or that need to be updated? I am still a bicep beginner and would be happy about an update.

**Comment by maf-soft** [comment]
Btw., I'm not sure if [package-lock.json](https://github.com/mariomeyrelles/bicep-functions-cosmos-keyvault/blob/main/package-lock.json) should be updated or removed/ignored from the repo. Just tell if you want me to push the updated one.

**Comment by mariomeyrelles** [comment]
I shouldn't have added it. Adding it to gitignore should be good.

**Comment by maf-soft** [comment]
I will add that to the pull request

**Comment by maf-soft** [comment]
I was nearly done adding that to the pull request, but then decided to google it just to be sure :)
https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5
According to [npm help package-lock.json](https://docs.npmjs.com/cli/v8/configuring-npm/package-lock-json) this file is intended to be committed. So I offered the updated file instead...

**Comment by mariomeyrelles** [comment]
I agree with your judgment. I will try to test the changes on your PR today or tomorrow, and then I will merge.

**Comment by maf-soft** [comment]
I guess you are busy? :)

I have another problem:
"DeploymentFailed - Tenant ID, application ID, principal ID, and scope are not allowed to be updated. (Code:RoleAssignmentUpdateNotPermitted)"

I guess that happens because I manually added another person and the template tries to correct this? How can I ignore such not permitted changes? Maybe this could be improved also in the public version.
